storage/src/test/scala/io/casperlabs/storage/dag/MessageAncestorsStorageTest.scala

```diff
@@ -0,0 +1,134 @@
+package io.casperlabs.storage.dag
+
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers
+import org.scalatest.compatible.Assertion
+import monix.eval.Task
+import monix.execution.Scheduler
+import cats.effect.laws.discipline.arbitrary
+import com.google.protobuf.ByteString
+import io.casperlabs.casper.consensus.Block
+import io.casperlabs.casper.consensus.Signature
+import io.casperlabs.storage.{ArbitraryStorageData, SQLiteFixture}
+import io.casperlabs.storage.SQLiteStorage
+import io.casperlabs.storage.block.BlockStorage
+import io.casperlabs.storage.BlockMsgWithTransform
+
+class MessageAncestorsStorageTest
+    extends FlatSpec
+    with Matchers
+    with ArbitraryStorageData
+    with SQLiteFixture[BlockStorage[Task] with MessageAncestorsStorage[Task]] {
+
+  override def db: String = "/tmp/message_ancestors_test.db"
+
+  override def createTestResource: Task[BlockStorage[Task] with MessageAncestorsStorage[Task]] =
+    SQLiteStorage.create[Task](readXa = xa, writeXa = xa)
+
+  implicit val consensusConfig: ConsensusConfig = ConsensusConfig(
+    dagSize = 5,
+    dagDepth = 3,
+    dagBranchingFactor = 1,
+    maxSessionCodeBytes = 1,
+    maxPaymentCodeBytes = 1,
+    minSessionCodeBytes = 1,
+    minPaymentCodeBytes = 1
+  )
+
+  def randomMessage: Block =
+    sample(arbBlock.arbitrary)
+
+  def createGenesis: Block = {
+    val b = randomMessage
+    b.update(
+        _.header := b.getHeader
+          .withParentHashes(Seq.empty)
+          .withValidatorPublicKey(ByteString.EMPTY)
+          .withMainRank(0)
+      )
+      .withSignature(Signature())
```

Comment:
```suggestion
      .clearSignature
```
Genesis is not signed, I think this is clearer. `b.withHeader` looks simpler as well.

Reply:
Done.